packages/meditrak-server/src/tests/routes/GETSurveyResponses/assertSurveyResponsesPermission.test.js

```diff
@@ -0,0 +1,144 @@
+ */
+
+import { expect } from 'chai';
+import { AccessPolicy } from '@tupaia/access-policy';
+import {
+  buildAndInsertSurveys,
+  buildAndInsertSurveyResponses,
+  findOrCreateDummyRecord,
+  generateTestId,
+} from '@tupaia/database';
+import { getModels } from '../../getModels';
+import {
+  filterSurveyResponsesByPermissions,
+  assertSurveyResponsePermissions,
+} from '../../../routes/GETSurveyResponses/assertSurveyResponsePermissions';
+
+describe('Permissions checker for GETSurveyResponses', async () => {
+  const DEFAULT_POLICY = {
+    VU: ['Admin', 'Donor'],
+    LA: ['Admin'],
+  };
+
+  const models = getModels();
+  let surveyResponses = [];
+  const laosAdminResponseId = generateTestId();
+
+  before(async () => {
+    const adminPermissionGroup = await findOrCreateDummyRecord(models.permissionGroup, {
+      name: 'Admin',
+    });
+    const donorPermissionGroup = await findOrCreateDummyRecord(models.permissionGroup, {
+      name: 'Donor',
+    });
+    const vanuatuCountry = await findOrCreateDummyRecord(models.country, { code: 'VU' });
+    const laosCountry = await findOrCreateDummyRecord(models.country, { code: 'LA' });
+    const vanuatuEntity = await findOrCreateDummyRecord(models.entity, {
+      id: generateTestId(),
+      code: 'VU_TEST',
+      country_code: vanuatuCountry.code,
+    });
+    const laosEntity = await findOrCreateDummyRecord(models.entity, {
+      id: generateTestId(),
+      code: 'LA_TEST',
+      country_code: laosCountry.code,
+    });
+
+    await buildAndInsertSurveys(models, [
+      {
+        id: generateTestId(),
+        code: 'TEST_SURVEY_1',
+        name: 'Test Survey 1',
+        permission_group_id: adminPermissionGroup.id,
+      },
+      {
+        id: generateTestId(),
+        code: 'TEST_SURVEY_2',
+        name: 'Test Survey 2',
+        permission_group_id: donorPermissionGroup.id,
+      },
+    ]);
+
+    const surveyResponseModels = await buildAndInsertSurveyResponses(models, [
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: laosAdminResponseId,
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: laosEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: laosEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+    ]);
+    surveyResponses = surveyResponseModels.map(srm => srm.surveyResponse);
+  });
+
+  describe('filterSurveyResponsesByPermissions()', async () => {
```

Comment:
Maybe add another test case that users have access to all the passed in survey responses and this function doesn't filter any thing?

Reply:
Also if users have BES Admin access to any countries, it should have access to all the survey responses without any restriction. So would be great if we can add test cases for that scenario too (I missed this in my code too, just added it recently)

Reply:
Added test cases for all permissions and for BES Admin permissions.